### PR TITLE
[WIP] Major Documentation Revamp

### PR DESCRIPTION
**Pull request by Ayushk4**
This PR is aimed at improving the docs in the following ways -
- Adding examples
- Make it easy to understand
- Improving the styling
- Fixing typos and grammatical errors

Also testing for the possible errors in the package.

**Progress**
- [x] Documents
    - [X] Creating documents
    - [x] Basic functions for working with Documents
    - [x] Document Metadata
    - [x] Preprocessing
- [x] Corpus
    - [x] Creating a Corpus
    - [x] Standardizing a Corpus
    - [x] Processing a Corpus
    - [x] Corpus Statistics
    - [x] Converting a DataFrame from a Corpus
    - [x] Corpus Metadata
- [x] Features
    - [x] Creating a Document Term Matrix
    - [x] Creating Individual Rows of a DTM
    - [x] Hash Trick
    - [x] TF-IDF
- [x] Semantic
    - [x] LSA
    - [x] LDA

### CODE REVIEW

**Comment by Ayushk4** [comment]
I am also replacing the deprecated functions in the document. So, far I have finished `documents.md`.

**Comment by Ayushk4** [comment]
@aviks I was going through the docs. I found that [`names`](https://github.com/JuliaText/TextAnalysis.jl/blob/b3c467cebb776c6a7e686cfe2b6c9a20c2ca2205/src/metadata.jl#L32) (used for setting metadata - author name for entire corpus of docs), conflicts with the base refer [(v0.7)](https://docs.julialang.org/en/v0.7/base/base/#Base.names) and [(v1)](https://docs.julialang.org/en/v1/base/base/#Base.names). What do you suggest about this? Should I open an issue?

**Comment by aviks** [comment]
`author` maybe?

**Comment by Ayushk4** [comment]
A correction - Author/authors is already used for document's author, should I change the metadata containing the document name to `documentName`. Also, should I send a separate PR for this?

**Comment by Ayushk4** [comment]
The following are the metadata - `names` -> setting the document's name 
and `author` for setting the author's name. 

This is also missing from the documentation, I am adding it.

**Comment by zgornel** [comment]
- `Documenter.jl` seems to be at this point a better choice since all code snippets are (re-)evaluated at branch modifications and reflect changes in the code. Is there any reason for not using it ?

 - A more flexible approach to the metadata field names is to allow a metadata type hierarchy (i.e. define an `AbstractMetadata` type) https://zgornel.github.io/StringAnalysis.jl/dev/doc_extensions/

**Comment by Ayushk4** [comment]
- I don't see any downside of using `Documenter.jl` features of `@repl` and doctests. Maybe this can be reflected across the documentation of JuliaText Repositories.

- A lot of things has to be ported from StringAnalysis.jl. I came across this issue - https://github.com/zgornel/StringAnalysis.jl/issues/2#issuecomment-451190789.

**Comment by Ayushk4** [comment]
I think that maybe we can start with opening an issue of what all major changes are to be done for TextAnalysis.jl taking into consideration the things mentioned in the conversation between you and aviks. Thank you, for your review on the situation.

**Comment by Ayushk4** [comment]
@aviks Can you please suggest your views on this, so that I may proceed.

File: docs/src/corpus.md [hunk 1]
```diff
@@ -4,35 +4,54 @@ Working with isolated documents gets boring quickly. We typically want to
...
+```julia
+julia> crps = Corpus([StringDocument("Document 1"),
+                      StringDocument("Document 2")])
+Corpus{StringDocument{String}}(StringDocument{String}[StringDocument{String}("Document 1", DocumentMetadata(English(), "Unnamed Document", "Unknown Author", "Unknown Time")), StringDocument{String}("Document 2", DocumentMetadata(English(), "Unnamed Document", "Unknown Author", "Unknown Time"))], 0, Dict{String,Int64}(), Dict{String,Array{Int64,1}}(), TextHashFunction(hash, 100))
+```
 
 ## Standardizing a Corpus
 
-A `Corpus` may contain many different types of documents:
+!!! note
```
Comment (aviks) [inline docs/src/corpus.md]: remove this. people read what they want, let's not try and force them to do something. this does not add anything to the docs.

File: docs/src/corpus.md [hunk 2]
```diff
@@ -83,3 +139,56 @@ corpus. The easiest way to do this is to convert a `Corpus` object into
 a `DataFrame`:
 
     convert(DataFrame, crps)
+
+## Corpus Metadata
+
+You can also retrieve the metadata for every document in a `Corpus` at once:
+
+* `languages()`: What language is the document in? Defaults to `Languages.English()`, a Language instance defined by the Languages package.
+* `names()`: What is the name of the document? Defaults to `"Unnamed Document"`.
```
Comment (aviks) [inline docs/src/corpus.md]: I think `names` should be changed to `titles`.  I was earlier unclear what `names` refers to.

**Comment by aviks** [comment]
Thanks Ayush, this looks good. Let me know once this is ready to merge. I'm happy to merge things over from StringAnalysis if Corneliu is ok with that.

**Comment by zgornel** [comment]
@aviks I'm perfectly fine with this; feel free to port to `TextAnalysis` whatever you see fit

File: docs/src/corpus.md (hunk 2 above)
Comment (Ayushk4) [reply docs/src/corpus.md]: Do you prefer that I send another PR making changes, `name` -> `title` & `name!` -> `title!` & `names` -> `titles` & `names!` to `titles!` (across the entire codebase, including tests, docs) ? Or include in this one itself?

**Comment by aviks** [comment]
Separate one please

File: docs/src/corpus.md (hunk 2 above)
Comment (Ayushk4) [reply docs/src/corpus.md]: I am marking this resolved as per #135

File: docs/src/corpus.md (hunk 1 above)
Comment (Ayushk4) [reply docs/src/corpus.md]: I am marking this resolved via the latest pushed commit.

**Comment by Ayushk4** [comment]
@aviks I am almost done with the PR. All that remains to be done, is #135 being merged so that I can remove the merge conflicts that may arise. 

Also, `lsa()` is not working. I will get more details about it and file a detailed issue.

**Comment by Ayushk4** [comment]
@aviks Do you think, BM25 and Co-occurrence matrix will be a good addition to TextAnalysis.jl?

**Comment by aviks** [comment]
> @aviks Do you think, BM25 and Co-occurrence matrix will be a good addition to TextAnalysis.jl?

Yes please!

**Pull request merged**